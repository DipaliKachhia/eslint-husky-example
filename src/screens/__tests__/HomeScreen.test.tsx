import React from 'react';
import '@testing-library/react-native/dont-cleanup-after-each';
import { render, screen, fireEvent, act, cleanup, waitFor } from '@testing-library/react-native';
import HomeScreen from '../HomeScreen';

describe('Home Screen', () => {
  let renderedComponent: any;
  beforeAll(() => {
    renderedComponent = render(<HomeScreen />);
  });
  afterAll(() => {
    cleanup();
  });
  test('should render', () => {
    act(() => {
      const { toJSON } = renderedComponent;
      expect(toJSON()).toMatchSnapshot();
    });
  });
  describe('When user open home screen', () => {
    test('then it should show Eslint example text', () => {
      expect(screen.getByText('EsLint Example')).toBeDefined();
    });
  });
});
