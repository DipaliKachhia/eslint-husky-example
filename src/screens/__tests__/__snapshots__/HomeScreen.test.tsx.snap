// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen should render 1`] = `
<RCTScrollView
  contentInsetAdjustmentBehavior="automatic"
  style={
    {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        [
          {
            "paddingBottom": 40,
            "paddingHorizontal": 32,
            "paddingTop": 96,
          },
          {
            "backgroundColor": "#F3F3F3",
          },
        ]
      }
    >
      <Image
        accessibilityRole="image"
        source={
          {
            "testUri": "../Libraries/NewAppScreen/components/logo.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            {
              "marginBottom": -192,
              "marginLeft": -128,
              "opacity": 0.2,
              "overflow": "visible",
              "resizeMode": "cover",
            },
          ]
        }
        testID="new-app-screen-header"
      />
      <Text
        style={
          [
            {
              "fontSize": 40,
              "fontWeight": "700",
              "textAlign": "center",
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Welcome to
        

        React Native
      </Text>
    </View>
    <View>
      <Text>
        EsLint Example
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
